src/utils/dates.js

```diff
@@ -17,3 +17,42 @@ export function getDaysBetweenDates(newDate, oldDate) {
 	const differenceInDays = Math.ceil(timeDifference / ONE_DAY_IN_MILLISECONDS);
 	return differenceInDays;
 }
+
+export function comparePurchaseUrgency(data) {
+	const classifiedByUrgency = data.map((item) => {
+		const dateLastPurchased =
+			item.dateLastPurchased === null
+				? new Date(item.dateCreated.toDate())
+				: new Date(item.dateLastPurchased.toDate());
+		const dateNextPurchased = new Date(item.dateNextPurchased.toDate());
```

Comment:
On these lines, you're already returning a date object by invoking `toDate()`, so you don't need `new Date()`.